source/kernel/trace_inode.c

```diff
@@ -0,0 +1,231 @@
+#include "trace_inode.h"
+
+#include <linux/atomic.h>
+#include <linux/dcache.h>
+#include <linux/errno.h>
+#include <linux/fs.h>
+#include <linux/hashtable.h>
+#include <linux/list.h>
+#include <linux/slab.h>
+#include "io_trace.h"
+#include "iotrace_event.h"
+#include "trace_env_kernel.h"
+
+#ifndef DEBUG
+#define DEBUG 0
+#endif
+
+#if 1 == DEBUG
+#define debug(format, ...)                                               \
+    printk(KERN_INFO "[iotrace][inode cache] %s " format "\n", __func__, \
+           ##__VA_ARGS__)
+#else
+#define debug(format, ...)
+#endif
+
+#define CACHE_ENTRIES 4096ULL
+#define HASH_ENTRIES CACHE_ENTRIES / 4ULL
+
```

Comment:
Could you provide additional file level documentation about the logic of caching inodes entries?